src/components/views/dialogs/spotlight/SpotlightDialog.tsx

```diff
@@ -555,7 +556,7 @@ const SpotlightDialog: React.FC<IProps> = ({ initialText = "", initialFilter = n
                         id={`mx_SpotlightDialog_button_result_${result.room.roomId}`}
                         key={`${Section[result.section]}-${result.room.roomId}`}
                         onClick={(ev) => {
-                            viewRoom(result.room.roomId, true, ev?.type !== "click");
+                            viewRoom({ roomId: result.room.roomId }, true, ev?.type !== "click");
```

Comment:
I wonder if TS would allow you to pass `result.room` as the room object here to make the code more readable